addons/mail/static/src/views/web/form/form_controller.js

```diff
@@ -54,9 +54,9 @@ patch(FormController.prototype, {
         if (
             changes.notification_type &&
             record.resModel === "res.users" &&
-            this.mailStore?.user.user.id === record.resId
+            this.mailStore?.self.user.id === record.resId
         ) {
-            this.mailStore.user.notification_preference = changes.notification_type;
+            this.mailStore.self.notification_preference = changes.notification_type;
```

Comment:
I think we should delete that code actually. Change of `notification_preference` should be sent on the bus from the python instead.

Reply:
This is weak as it only works on the current tab and for the current user.

Reply:
Unrelated to diff, but `updateUserNotificationPreference` method looks unused